strategy/compose_strategy.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018-2020 Gamebridge.ai authors. All rights reserved. MIT license.
+import { JSONValue } from "../serializable.ts";
+
+/** Functions used when hydrating data */
+export type FromJSONStrategy = (value: JSONValue) => any;
+export type FromJSONStrategyArgument =
+  (FromJSONStrategy | FromJSONStrategy[])[];
+
+/** Functions used when dehydrating data */
+export type ToJSONStrategy = (value: any) => JSONValue;
+export type ToJSONStrategyArgument = (ToJSONStrategy | ToJSONStrategy[])[];
+/** Function to build a `fromJSONStrategy` or `toJSONStrategy`.
+ * Converts value from functions provided as parameters
+ */
+export function composeStrategy(
+  ...fns:
+    | FromJSONStrategyArgument
+    | ToJSONStrategyArgument
```

Comment:
The way this is written you could technically chain both `from` and `to` strategies together
```ts
composeStrategy(toJSONDefault, fromJSONDefault);
```

Not sure if this would be an issue as I imagine this would mostly be used for values that are subclasses of both anyways (like `string` -> `string`)

Reply:
ya, It's okay, the way it was typed before was basically the same thing